discretize/BaseMesh.py

```diff
@@ -1,41 +1,57 @@
 import numpy as np
+import properties
+import os
+import json
 from discretize import utils
 
 
-class BaseMesh(object):
-    """BaseMesh does all the counting you don't want to do.
+class BaseMesh(properties.HasProperties):
+    """
+    BaseMesh does all the counting you don't want to do.
     BaseMesh should be inherited by meshes with a regular structure.
-
-    :param numpy.array n: (or list) number of cells in each direction (dim, )
-    :param numpy.array x0: (or list) Origin of the mesh (dim, )
-
     """
 
+    # Properties
+    n = properties.Array(
```

Comment:
Can we change this later? Having it as a public property seems to suggest that we can change it....

Reply:
see aranzgeo/properties#191

Reply:
agreed though - ideally, this should remain a private variable